src/root/views/DrefApplication/useDrefFormOptions.ts

```diff
@@ -0,0 +1,450 @@
+import React from 'react';
+import {
+  isDefined,
+} from '@togglecorp/fujs';
+import {
+  PartialForm,
+  ObjectSchema,
+} from '@togglecorp/toggle-form';
+
+import { compareString } from '#utils/utils';
+import LanguageContext from '#root/languageContext';
+import {
+  useRequest,
+  ListResponse,
+} from '#utils/restRequest';
+
+import {
+  requiredCondition,
+  requiredStringCondition,
+  positiveIntegerCondition,
+} from '#utils/form';
+
+import {
+  Country
+} from '#types';
+
+import {
+  BooleanValueOption,
+  NumericValueOption,
+  StringValueOption,
+  ReportType,
+  emptyNumericOptionList,
+  emptyActionList,
+  FormType,
+  Entity,
+  ActionFields,
+  ActionByReportType,
+  ActionsByOrganization,
+  OrganizationType,
+  STATUS_EARLY_WARNING,
+  STATUS_EVENT,
+  DISASTER_TYPE_EPIDEMIC,
+  BULLETIN_PUBLISHED_NO,
+  BULLETIN_PUBLISHED_PLANNED,
+  BULLETIN_PUBLISHED_YES,
+  SOURCE_RC,
+  SOURCE_GOV,
+  SOURCE_OTHER,
+  User,
+} from './common';
+
+type FormSchema = ObjectSchema<PartialForm<FormType>>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const getRequiredWithCondition = (key: keyof FormType) => (
+  value: number | string | null | undefined,
+  allValue: PartialForm<FormType>
+) => {
+  if (!isDefined(value) && isDefined(allValue?.[key])) {
+    return 'This field is required';
+  }
+
+  return undefined;
+};
+
+const getRequiredWithNonEmptyCondition = (key: keyof FormType) => (
+  value: number | string | null | undefined,
+  allValue: PartialForm<FormType>
+) => {
+  if (!value && allValue?.[key]) {
+    return 'This field is required';
+  }
+
+  return undefined;
+};
+
+const validStatusCondition = (value: number | string | null | undefined) => {
+  if (value === STATUS_EARLY_WARNING || value === STATUS_EVENT) {
+    return undefined;
+  }
+
+  return 'Status should either be an Event or an Early Warning / Early Action';
+};
```

Comment:
We may already have blacklist and whitelist condition.
But the error message is not as customizable.